pennylane/ops/functions/equal.py

```diff
@@ -183,9 +183,8 @@ def _equal_operators(
         return False
 
     if op1.arithmetic_depth > 0:
-        raise NotImplementedError(
-            "Comparison of operators with an arithmetic depth larger than 0 is not yet implemented."
-        )
+        # Comparison of operators with an arithmetic depth larger than 0 is not yet implemented.
+        return False
```

Comment:
this feels a bit strange and perhaps we should consider how to implement it, but I almost want to continue raising an error here. We know that they have the same arithmetic depth. They might actually be the same! I'm ok with this decision, but it's worth noting.

Reply:
I think this is a pretty rare occurrence at this point anyway, if that matters. This error used to be raised frequently, but those cases are now mostly (if not entirely) handled by other dispatches of the function (and these do use recursion, so they can handle the arithmetic depth). 

I'm not sure there are any operators with arithmetic depth greater than 0 that default to this dispatch. The test for this line runs by comparing two instances of `SymbolicOp(qml.PauliY(0))`, because I couldn't come up with an actual PennyLane operator that would raise this error now that we've implemented all the logic for the various subclasses of `CompositeOp` and `SymbolicOp` and `Controlled`. But maybe I just wasn't creative enough.

Reply:
So I think Line 294/ 289 was the main place I had in mind for this PR.  

Given all arithmetic ops *are* now implemented, I'm ambivalent about how we want to handle this logic.

Reply:
I think it's still worth returning `False` in the dispatcher that checks arithmetic depth. Not sure if we should return `False` or raise an error if none of the dispatches are used.